Based partially on Fedora patchset

--- origsrc/dcmtk-3.6.0/CMakeLists.txt	2014-11-20 10:32:02.884230700 -0600
+++ src/dcmtk-3.6.0/CMakeLists.txt	2014-11-20 10:30:00.829617900 -0600
@@ -20,12 +20,21 @@ SET(DCMTK_MODULES config ofstd oflog dcm
 SET(DCMTK_PACKAGE_NAME "dcmtk")
 SET(DCMTK_PACKAGE_DATE "2011-01-06")
 #SET(DCMTK_PACKAGE_TARNAME "dcmtk-3.6.0")
-SET(DCMTK_PACKAGE_VERSION "3.6.0")
-SET(DCMTK_PACKAGE_VERSION_NUMBER "360")
+#SET(DCMTK_PACKAGE_VERSION "3.6.0")
+#SET(DCMTK_PACKAGE_VERSION_NUMBER "360")
 SET(DCMTK_PACKAGE_VERSION_SUFFIX "")
 #SET(DCMTK_PACKAGE_STRING "dcmtk 3.6.0")
 #SET(DCMTK_PACKAGE_BUGREPORT "dicom-bugs@offis.de")
 
+SET(DCMTK_MAJOR_VERSION 3)
+SET(DCMTK_MINOR_VERSION 6)
+SET(DCMTK_BUILD_VERSION 0)
+SET(DCMTK_PACKAGE_VERSION "${DCMTK_MAJOR_VERSION}.${DCMTK_MINOR_VERSION}.${DCMTK_BUILD_VERSION}")
+SET(DCMTK_API_VERSION "${DCMTK_MAJOR_VERSION}.${DCMTK_MINOR_VERSION}")
+SET(DCMTK_LIBRARY_PROPERTIES VERSION "${DCMTK_PACKAGE_VERSION}" SOVERSION "${DCMTK_API_VERSION}")
+SET(DCMTK_PACKAGE_VERSION_NUMBER ${DCMTK_MAJOR_VERSION}${DCMTK_MINOR_VERSION}${DCMTK_BUILD_VERSION})
+
+
 # DCMTK build options
 OPTION(DCMTK_WITH_TIFF "Configure DCMTK with support for TIFF" ON)
 OPTION(DCMTK_WITH_PNG "Configure DCMTK with support for PNG" ON)
@@ -210,7 +219,7 @@ ELSE(WIN32)   # ... for non-Windows syst
     SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_XOPEN_SOURCE_EXTENDED -D_BSD_SOURCE -D_BSD_COMPAT -D_OSF_SOURCE -D_DARWIN_C_SOURCE")
   ELSE(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     # FreeBSD fails with these flags
-    IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+    IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD" AND NOT CYGWIN)
       # Compiler flags for all other non-Windows systems
       SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_XOPEN_SOURCE_EXTENDED -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -D_BSD_COMPAT -D_OSF_SOURCE -D_POSIX_C_SOURCE=199506L")
       SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_XOPEN_SOURCE_EXTENDED -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -D_BSD_COMPAT -D_OSF_SOURCE -D_POSIX_C_SOURCE=199506L")
@@ -295,8 +304,28 @@ ENDIF(WITH_THREADS)
 # Recurse into subdirectories
 FOREACH(module ${DCMTK_MODULES})
   ADD_SUBDIRECTORY(${module})
+  
+# for each shared lib add proper soname information
+    IF(BUILD_SHARED_LIBS)
+	IF(${module} MATCHES config)
+        # Skip config module
+	ELSEIF(${module} MATCHES doxygen)
+        # Skip doxygen module
+        ELSE(${module} MATCHES config)
+          SET_TARGET_PROPERTIES(${module} PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+        ENDIF(${module} MATCHES config)
+    ENDIF(BUILD_SHARED_LIBS)
+    
 ENDFOREACH(module)
 
+# Set target properties for modules which are not in DCMTK_MODULE
+SET_TARGET_PROPERTIES(dcmdsig PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+SET_TARGET_PROPERTIES(ijg12 PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+SET_TARGET_PROPERTIES(ijg16 PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+SET_TARGET_PROPERTIES(ijg8 PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+SET_TARGET_PROPERTIES(libi2d PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+SET_TARGET_PROPERTIES(charls PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+
 #-----------------------------------------------------------------------------
 # Installation tasks
 #-----------------------------------------------------------------------------
--- origsrc/dcmtk-3.6.0/dcmdata/libi2d/CMakeLists.txt	2014-11-20 10:32:02.884230700 -0600
+++ src/dcmtk-3.6.0/dcmdata/libi2d/CMakeLists.txt	2014-11-20 10:30:00.907618100 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(libi2d i2d i2dplvlp i2djpgs i2dbmps i2dplsc i2dplnsc)
 
+TARGET_LINK_LIBRARIES(libi2d dcmdata ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} libi2d)
--- origsrc/dcmtk-3.6.0/dcmdata/libsrc/CMakeLists.txt	2014-11-20 10:32:02.884230700 -0600
+++ src/dcmtk-3.6.0/dcmdata/libsrc/CMakeLists.txt	2014-11-20 10:30:00.907618100 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmdata cmdlnarg dcbytstr dcchrstr dccodec dcdatset dcddirif dcdicdir dcdicent dcdict dcdictzz dcdirrec dcelem dcerror dcfilefo dchashdi dcistrma dcistrmb dcistrmf dcistrmz dcitem dclist dcmetinf dcobject dcostrma dcostrmb dcostrmf dcostrmz dcpcache dcpixel dcpixseq dcpxitem dcrleccd dcrlecce dcrlecp dcrledrg dcrleerg dcrlerp dcsequen dcstack dcswap dctag dctagkey dctypes dcuid dcwcache dcvm dcvr dcvrae dcvras dcvrat dcvrcs dcvrda dcvrds dcvrdt dcvrfd dcvrfl dcvris dcvrlo dcvrlt dcvrobow dcvrof dcvrpn dcvrpobw dcvrsh dcvrsl dcvrss dcvrst dcvrtm dcvrui dcvrul dcvrulup dcvrus dcvrut dcxfer dcpath vrscan vrscanl)
 
+TARGET_LINK_LIBRARIES(dcmdata oflog ofstd ${ZLIB_LIBRARY})
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmdata)
--- origsrc/dcmtk-3.6.0/dcmimage/libsrc/CMakeLists.txt	2014-11-20 10:32:02.899830800 -0600
+++ src/dcmtk-3.6.0/dcmimage/libsrc/CMakeLists.txt	2014-11-20 10:30:00.938818100 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmimage diargimg dicmyimg dicoimg dicoopx dicopx dihsvimg dilogger dipalimg dipipng dipitiff diqtctab diqtfs diqthash diqthitl diqtpbox diquant diregist dirgbimg diybrimg diyf2img diyp2img)
 
+TARGET_LINK_LIBRARIES(dcmimage dcmimgle oflog ofstd ${PNG_LIBRARY_RELEASE} ${TIFF_LIBRARY})
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmimage)
--- origsrc/dcmtk-3.6.0/dcmimgle/libsrc/CMakeLists.txt	2014-11-20 10:32:02.899830800 -0600
+++ src/dcmtk-3.6.0/dcmimgle/libsrc/CMakeLists.txt	2014-11-20 10:30:00.985618200 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmimgle dcmimage dibaslut diciefn dicielut didislut didispfn didocu digsdfn digsdlut diimage diinpx diluptab dimo1img dimo2img dimoimg dimoimg3 dimoimg4 dimoimg5 dimomod dimoopx dimopx diovdat diovlay diovlimg diovpln diutils)
 
+TARGET_LINK_LIBRARIES(dcmimgle dcmdata oflog ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmimgle)
--- origsrc/dcmtk-3.6.0/dcmjpeg/libsrc/CMakeLists.txt	2014-11-20 10:32:02.899830800 -0600
+++ src/dcmtk-3.6.0/dcmjpeg/libsrc/CMakeLists.txt	2014-11-20 10:30:00.985618200 -0600
@@ -4,5 +4,7 @@ INCLUDE_DIRECTORIES(${dcmjpeg_SOURCE_DIR
 # create library from source files
 ADD_LIBRARY(dcmjpeg ddpiimpl dipijpeg djcodecd djcodece djcparam djdecbas djdecext djdeclol djdecode djdecpro djdecsps djdecsv1 djdijg12 djdijg8 djdijg16 djeijg12 djeijg8 djeijg16 djencbas djencext djenclol djencode djencpro djencsps djencsv1 djrplol djrploss djutils)
 
+TARGET_LINK_LIBRARIES(dcmjpeg ijg8 ijg12 ijg16 dcmimgle dcmdata ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmjpeg)
--- origsrc/dcmtk-3.6.0/dcmjpls/libsrc/CMakeLists.txt	2014-11-20 10:32:02.899830800 -0600
+++ src/dcmtk-3.6.0/dcmjpls/libsrc/CMakeLists.txt	2014-11-20 10:30:00.985618200 -0600
@@ -4,5 +4,7 @@ INCLUDE_DIRECTORIES(${dcmjpls_SOURCE_DIR
 # create library from source files
 ADD_LIBRARY(dcmjpls djcparam djdecode djencode djrparam djcodecd djutils djcodece)
 
+TARGET_LINK_LIBRARIES(dcmjpls charls dcmimgle dcmdata oflog ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmjpls)
--- origsrc/dcmtk-3.6.0/dcmnet/libsrc/CMakeLists.txt	2014-11-20 10:32:02.899830800 -0600
+++ src/dcmtk-3.6.0/dcmnet/libsrc/CMakeLists.txt	2014-11-20 10:30:00.985618200 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmnet assoc cond dcasccff dcasccfg dccfenmp dccfpcmp dccfprmp dccfrsmp dccftsmp dccfuidh dcmlayer dcmtrans dcompat dimcancl dimcmd dimdump dimecho dimfind dimget dimmove dimse dimstore diutil dul dulconst dulextra dulfsm dulparse dulpres extneg lst dfindscu dcuserid scu scp)
 
+TARGET_LINK_LIBRARIES(dcmnet dcmdata oflog ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmnet)
--- origsrc/dcmtk-3.6.0/dcmpstat/libsrc/CMakeLists.txt	2014-11-20 10:32:02.899830800 -0600
+++ src/dcmtk-3.6.0/dcmpstat/libsrc/CMakeLists.txt	2014-11-20 10:30:00.000000000 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmpstat dcmpstat dviface dvpsab dvpsabl dvpsal dvpsall dvpscf dvpscu dvpscul dvpsda dvpsdal dvpsfs dvpsga dvpsgal dvpsgl dvpsgll dvpsgr dvpsgrl dvpshlp dvpsib dvpsibl dvpsmsg dvpsov dvpsovl dvpspl dvpspl2 dvpspll dvpspr dvpsprt dvpsri dvpsril dvpsrs dvpsrsl dvpssp dvpsspl dvpssv dvpssvl dvpstat dvpstx dvpstxl dvpsvl dvpsvll dvpsvw dvpsvwl dvsighdl)
 
+TARGET_LINK_LIBRARIES(dcmpstat dcmdsig dcmtls dcmsr dcmqrdb dcmnet dcmimgle dcmdata oflog ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmpstat)
--- origsrc/dcmtk-3.6.0/dcmqrdb/libsrc/CMakeLists.txt	2014-11-20 10:32:02.915430800 -0600
+++ src/dcmtk-3.6.0/dcmqrdb/libsrc/CMakeLists.txt	2014-11-20 10:30:00.000000000 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmqrdb dcmqrcbf dcmqrcbg dcmqrcbm dcmqrcbs dcmqrcnf dcmqrdbi dcmqrdbs dcmqropt dcmqrptb dcmqrsrv dcmqrtis)
 
+TARGET_LINK_LIBRARIES(dcmqrdb dcmnet dcmdata oflog ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmqrdb)
--- origsrc/dcmtk-3.6.0/dcmsign/libsrc/CMakeLists.txt	2014-11-20 10:32:02.915430800 -0600
+++ src/dcmtk-3.6.0/dcmsign/libsrc/CMakeLists.txt	2014-11-20 10:30:00.000000000 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmdsig dcsignat siautopr sibrsapr sicert sicertvf sicreapr sidsa simaccon simd5 sinullpr siprivat siripemd sirsa sisha1 sisprof sitypes)
 
+TARGET_LINK_LIBRARIES(dcmdsig dcmdata oflog ofstd ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY})
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmdsig)
--- origsrc/dcmtk-3.6.0/dcmsr/libsrc/CMakeLists.txt	2014-11-20 10:32:02.915430800 -0600
+++ src/dcmtk-3.6.0/dcmsr/libsrc/CMakeLists.txt	2014-11-20 10:30:00.000000000 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmsr dsrcitem dsrcodtn dsrcodvl dsrcomtn dsrcomvl dsrcontn dsrcsidl dsrdattn dsrdoc dsrdoctn dsrdoctr dsrdtitn dsrimgfr dsrimgtn dsrimgvl dsrnumtn dsrnumvl dsrpnmtn dsrreftn dsrscogr dsrsc3gr dsrscotn dsrsc3tn dsrscovl dsrsc3vl dsrsoprf dsrstrvl dsrtcodt dsrtcosp dsrtcotn dsrtcoto dsrtcovl dsrtextn dsrtimtn dsrtncsr dsrtree dsrtypes dsruidtn dsrwavch dsrwavtn dsrwavvl dsrxmlc dsrxmld dsriodcc dsrbascc dsrenhcc dsrcomcc dsrkeycc dsrmamcc dsrchecc dsrcolcc dsrprocc dsrxrdcc dsrspecc dsrmaccc dsrimpcc)
 
+TARGET_LINK_LIBRARIES(dcmsr dcmdata oflog ofstd ${LIBXML2_LIBRARIES})
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmsr)
--- origsrc/dcmtk-3.6.0/dcmtls/libsrc/CMakeLists.txt	2014-11-20 10:32:02.915430800 -0600
+++ src/dcmtk-3.6.0/dcmtls/libsrc/CMakeLists.txt	2014-11-20 10:30:00.000000000 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmtls tlslayer tlstrans tlsscu)
 
+TARGET_LINK_LIBRARIES(dcmtls dcmnet ofstd ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY})
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmtls)
--- origsrc/dcmtk-3.6.0/dcmwlm/libsrc/CMakeLists.txt	2014-11-20 10:32:02.915430800 -0600
+++ src/dcmtk-3.6.0/dcmwlm/libsrc/CMakeLists.txt	2014-11-20 10:30:00.000000000 -0600
@@ -1,5 +1,7 @@
 # create library from source files
 ADD_LIBRARY(dcmwlm wlds wldsfs wlfsim wlmactmg)
 
+TARGET_LINK_LIBRARIES(dcmwlm dcmnet oflog ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} dcmwlm)
--- origsrc/dcmtk-3.6.0/oflog/libsrc/CMakeLists.txt	2014-11-20 10:32:02.915430800 -0600
+++ src/dcmtk-3.6.0/oflog/libsrc/CMakeLists.txt	2014-11-20 10:30:00.000000000 -0600
@@ -8,5 +8,7 @@ ENDIF(WIN32 AND NOT CYGWIN)
 
 ADD_LIBRARY(oflog oflog apndimpl appender config consap factory fileap filter globinit hierarchy hierlock layout logger logimpl logevent loglevel loglog lloguser ndc ntelogap nullap objreg patlay pointer property rootlog sleep socketap sockbuff socket strhelp syncprims syslogap threads timehelp ${OFLOG_PLATFORM_LIBRARIES})
 
+TARGET_LINK_LIBRARIES(oflog ofstd)
+
 # declare installation files
 INSTALL_TARGETS(${INSTALL_LIBDIR} oflog)
